File: 1160-find-words-that-can-be-formed-by-characters/1160-find-words-that-can-be-formed-by-characters.py
class Solution:
    def countCharacters(self, words: List[str], chars: str) -> int:
        char_count = Counter(chars)
        total = 0

        for w in words:
            word_count = Counter(w)
            if all(word_count[c] <= char_count[c] for c in word_count):
                total += len(w)

        return total

File: 1160-find-words-that-can-be-formed-by-characters/1160-find-words-that-can-be-formed-by-characters.go
func countCharacters(words []string, chars string) int {
    charCount := make(map[rune]int)
    for _, c := range chars {
        charCount[c]++
    }

    totalLength := 0
    for _, word := range words {
        wordCount := make(map[rune]int)
        for _, c := range word {
            wordCount[c]++
        }

        valid := true
        for c, count := range wordCount {
            if count > charCount[c] {
                valid = false
                break
            }
        }

        if valid {
            totalLength += len(word)
        }
    }

    return totalLength
}